context/src/main/java/io/grpc/Deadline.java

```diff
@@ -164,13 +163,40 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
-  /** Time source representing nanoseconds since fixed but arbitrary point in time. */
-  abstract static class Ticker {
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof Deadline)) {
+      return false;
+    }
+    Deadline that = (Deadline) obj;
+    return this.deadlineNanos == that.deadlineNanos
+        && this.ticker == that.ticker
+        && this.expired == that.expired;
```

Comment:
Don't compare expired. One instance may not have noticed it is actually expired yet.